source/extensions/filters/http/cache/cache_filter.cc

```diff
@@ -39,11 +39,12 @@ CacheFilter::CacheFilter(const envoy::extensions::filters::http::cache::v3::Cach
 
 void CacheFilter::onDestroy() {
   filter_state_ = FilterState::Destroyed;
-  if (lookup_) {
+  if (lookup_ != nullptr) {
```

Comment:
for risk I consider anything with buffering and watermarks to be inherently high risk.  IMO we should flag protect this and we should definitely call it out in release notes.

Reply:
The cache filter is still marked WIP and not production ready. If you think it needs this guardrail despite that, can you point me at a good example of a flag protected change? I'm not really sure how to do it.

One question for you that might be an actual danger here - if the filter is being destroyed while a high watermark state is active, does a low watermark event need to be sent? (Right now this does not happen, it would need to happen in `setSelfOwned` as the last moment when the relevant callback is still available.)

Reply:
mmm, given it's marked WIP I guess this is fine?  I defer to @jmarantz who didnt' ask for it so if he doesn't weigh in you can mark this resolved.

Reply:
If the filter is not supposed to be production-ready then I agree a runtime feature is not needed.